### ISSUE TITLE
`invalid-envvar-value` does not support StrEnum

### ISSUE THREAD

**Issue by stdedos** [PyCQA/pylint#10840]
### Bug description

```python
import os

from enum import StrEnum


class Env(StrEnum):
    """Env"""

    LOG_LEVEL = "LOG_LEVEL"
    """Set the log level"""


# pylint: disable-next=invalid-envvar-name # Probably false-positive: https://github.com/pylint-dev/pylint/issues/10840
os.getenv(Env.LOG_LEVEL, "INFO")
```

### Configuration

```ini

```

### Command used

```shell
uv run pylint a.py
```

### Pylint output

```python
************* Module a
a.py:13:0: W0012: Unknown option value for 'disable-next', expected a valid pylint message and got 'invalid-envvar-name' (unknown-option-value)
a.py:14:0: E1507: os.getenv does not support a.Env.LOG_LEVEL type argument (invalid-envvar-value)

-----------------------------------
Your code has been rated at 0.00/10

```

### Expected behavior

_Nothing_. `StrEnum` or `<enum>.name` resolve to `str`, which are valid type arguments

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.3
Python 3.12.3 (main, Jan  8 2026, 11:30:50) [GCC 13.3.0]
```

### Additional dependencies

```python

```

**Comment by stdedos** [comment]
... I couldn't decide if https://github.com/pylint-dev/pylint/issues/5091 is a duplicate or related - so I am just mentioning it here

**Comment by stdedos** [comment]
~~... is `disable-next` broken too??~~ _I guess what `Unknown option value for 'disable', expected a valid pylint message and got 'invalid-envvar-name'` is trying to tell me is: I cannot copy_ 🤦

```py
a.py:13:0: W0012: Unknown option value for 'disable-next', expected a valid pylint message and got 'invalid-envvar-name' (unknown-option-value)
```

My usage seems to match the https://pylint.pycqa.org/en/latest/user_guide/messages/message_control.html#block-disables example

**Comment by mbyrnepr2** [comment]
```python
# pylint: disable-next=invalid-envvar-value
...
```

**Comment by Pierre-Sassoulas** [comment]
Already said it here: https://github.com/pylint-dev/pylint/issues/5091#issuecomment-1176539836 but as we had a new bug opened I think this message does not have a place in pylint (a type checker does it better and having to maintain it feel like a useless chore) and I would favor either removing it or moving it to an external plugin (if we do an external plugin it might be the occasion to do a cookie cutter pylint plugin template, but I'm really under the water atm).
